modules/nwtc-library/src/NWTC_IO.f90

```diff
@@ -2337,18 +2331,21 @@ SUBROUTINE OpenBInpFile ( Un, InFile, ErrStat, ErrMsg )
       RETURN
    END IF
 
-
       ! Open input file.  Make sure it worked.
-   OPEN( Un, FILE=TRIM( InFile ), STATUS='OLD', FORM='UNFORMATTED', ACCESS='STREAM', IOSTAT=ErrStat, ACTION='READ' )
-
-   IF ( ErrStat /= 0 ) THEN
-      ErrStat = ErrID_Fatal
-      ErrMsg  = 'OpenBInpFile:Cannot open file "'//TRIM( InFile )//'" for reading. Another program may have locked it.'
-   ELSE
-      ErrStat = ErrID_None
-      ErrMsg  = ''
-   END IF
-
+   !$OMP critical
+   call GetNewUnit(Un,ErrStat,ErrMsg)
```

Comment:
For backward compatibility, could you wrap this call to GetNewUnit (here and the other places it's now called) in an IF statement like IF (Un < 0) call GetNewUnit()?

Reply:
Alternatively,  `GetNewUnit` could be changed to start looking at units at the given `Un` value if it's >= 0. Otherwise, `Un` is really `intent(out)` only.

Reply:
I'll wrap those calls in an `if` statement.  I think we want to maintain that backwards compatibility for now.